src/core/eval.ml

```diff
@@ -122,7 +133,7 @@ and eq_modulo : term -> term -> bool = fun a b ->
     | (Kind       , _          )
     | (_          , Kind       ) -> assert false
     | (Type       , Type       ) -> eq_modulo l
-    | (Vari(x1)   , Vari(x2)   ) when Bindlib.eq_vars x1 x2 -> eq_modulo l
+    | (Vari(x)    , Vari(y)    ) when Bindlib.eq_vars x y -> eq_modulo l
```

Comment:
There is something missing here. You should test variable unfolding. The code for variables should be as follows:
```
| (Vari(x)    , Vari(y)    ) when Bindlib.eq_vars x y -> eq_modulo l
| (Vari(x)    , t            )
| (t              , Vari(x)  ) ->
  begin
    match def_of ctx x with
    | None      -> false
    | Some(u) -> eq_modulo ((t,u)::l)
  end
```

Reply:
I think this case will be unused since `whnf` unfolds defined variables

Reply:
My previous assertion will likely be true if the context is added to all `whnf` calls.

Reply:
I see.